Cryptography API docs (#3319)
dapr/docs: daprdocs/content/en/reference/api/cryptography_api.md

 ```
| Setting | Description |
| ------- | ----------- |
| `metadata.name` | The name of the workflow component. |
| `spec/metadata` | Additional metadata parameters specified by workflow component |

## Supported workflow methods

Todo: organize the following into the correct format for an API doc.

The new building block would feature 7 APIs:

/encrypt: encrypts arbitrary data using a key stored in the vault. It supports symmetric and asymmetric ciphers, depending on the type of key in use (and the types of keys supported by the vault).
/decrypt: decrypts arbitrary data, performing the opposite of what /encrypt does.
/wrapkey: wraps keys using other keys stored in the vault. This is exactly like encrypting data, but it expects inputs to be formatted as keys (for example formatted as JSON Web Key) and it exposes additional algorithms not available when encrypting general data (like AES-KW)
/unwrapkey: un-wraps (decrypts) keys, performing the opposite of what /wrap does
/sign: signs an arbitrary message using an asymmetric key stored in the vault (we could also consider offering HMAC here, using symmetric keys, although not widely supported by the vault services)

> /getkey: this can be used only with asymmetric keys stored in the vault, and returns the public part of the key

Reviewer: this is not going to be part of 1.11. 
Only the high level APIs will be available.

---
type: docs
title: "Cryptography API reference"
linkTitle: "Cryptography API"
description: "Detailed documentation on the cryptography API"
weight: 900
---

## Component format

A Dapr `crypto.yaml` component file has the following structure:

```yaml
apiVersion: dapr.io/v1alpha1
kind: Component
metadata:
  name: <NAME>
spec:
  type: crypto.<TYPE>
  version: v1.0-alpha1
  metadata:
  - name: <NAME>
    value: <VALUE>
 ```

| Setting | Description |
| ------- | ----------- |
| `metadata.name` | The unique name of the workflow component. |
| `spec.type` | The component type used. Example: `crypto.jwks`, `crypto.azure.keyvault` |
| `spec.metadata` | Additional metadata parameters specified by workflow component |

[Learn more about the available cryptography components.]({{< ref supported-cryptography >}})

## Supported cryptography APIs

> The cryptography building block supports two high-level APIs:

Reviewer 2: ```suggestion
The cryptography building block supports two APIs:
```

- `Encrypt` 
- `Decrypt` 

> These APIs allow you to encrypt and decrypt files of arbitrary lenght (up to 256TB) while working on a straem of data.

Reviewer 2: ```suggestion
These APIs allow you to encrypt and decrypt files of arbitrary length (up to 256TB) while working on a stream of data.
```

### Encrypt

> To encrypt data, implement the `Encrypt` API:

Reviewer 2: ```suggestion
To encrypt data, use the `Encrypt` API:
```

```go
// Encrypt the data using Dapr

> out, err := sdkClient.Encrypt(context.Background(), rf, dapr.EncryptOptions{

Reviewer 2: This is the HTTP API definition, so currently there is no crypto HTTP API? If that is that case then we should remove this section (have no API doc) and simply say in the crypto overview that only gRPC is available in the alpha release

Author: @ItalyPaleAle thoughts?

	// These are the 3 required parameters
	ComponentName: "mycryptocomponent",
	KeyName:        "mykey",
	Algorithm:     "RSA",
})
```

### Decrypt

To decrypt data, implement the `Decrypt` API:

```go
// Decrypt the data using Dapr

> out, err := sdkClient.Decrypt(context.Background(), rf, dapr.EncryptOptions{

Reviewer 2: Remove this whole file since on gRPC

Author: @ItalyPaleAle I'll go ahead and remove this api reference doc, let me know if I should put back.
